components/ILIAS/OrgUnit/classes/Settings/class.ilOrgUnitObjectPositionSetting.php

```diff
@@ -16,93 +17,109 @@
  ********************************************************************
  */
 
-/* Copyright (c) 1998-2010 ILIAS open source, Extended GPL, see docs/LICENSE */
+declare(strict_types=1);
 
 /**
- * Stores object activation status of orgunit position settings.
- * @author Stefan Meyer <smeyer.ilias@gmx.de>
+ * Object-specific activation settings for orgunit position.
  */
 class ilOrgUnitObjectPositionSetting
 {
     protected ilDBInterface $db;
-    private int $obj_id;
     private ?bool $active = null;
+    private ?ilOrgUnitObjectTypePositionSetting $settings_cache = null;
 
-    public function __construct(int $a_obj_id)
-    {
+    public function __construct(
+        private int $obj_id
+    ) {
         $this->db = $GLOBALS['DIC']->database();
-        $this->obj_id = $a_obj_id;
         $this->readSettings();
     }
 
-    /**
-     * Lookup activation status
-     */
-    public function lookupActive(int $a_obj_id): bool
+    public static function getFor(int $obj_id): self
```

Comment:
Please add static caching for this method. We will have a bunch of lookups into orgu settings per object, so this shall prove beneficial for performance.